### PR TITLE
Send several NO_OLM in one request

### PR DESCRIPTION
**Pull request by BillCarsonFr**
NO_OLM withheld code were sent one by one during keyshare process.
In big room this can be a common scenario (people running out of OTK), this change simply send all the NO_OLM in a single toDevice event.

With a test account in Megolm Test room, this tear down the number of needed requests to 15 instead of 835

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/algorithms/megolm/MXMegolmEncryption.kt [hunk 1]
```diff
@@ -304,6 +301,16 @@ internal class MXMegolmEncryption(
         } else {
             Timber.i("## CRYPTO | shareUserDevicesKey() : no need to sharekey")
         }
+
+        if (noOlmToNotify.isNotEmpty()) {
+            //XXX offload?, as they won't read the message anyhow?
+            notifyKeyWithHeld(
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/algorithms/megolm/MXMegolmEncryption.kt:307]: This is now sent after the code with `if (haveTargets)`, I think it is OK, but could you confirm please?

**Review by bmarty — APPROVED** [review]
Thanks, LGTM.
2 minor remarks.
Also you should be interested to limit the size of the logs by using `logLimit()` in some of the logs in this class.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/algorithms/megolm/MXMegolmEncryption.kt (hunk 1 above)
Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/algorithms/megolm/MXMegolmEncryption.kt:307]: yes that's what i wanted
(i even wonder if we could offload to other thread to speed up sharing to the one that will be able to read the message)

Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/algorithms/megolm/MXMegolmEncryption.kt:307]: ok, thanks

**Pull request merged**